src/systemctl/systemctl-logind.c

```diff
@@ -75,6 +80,32 @@ int logind_reboot(enum action a) {
         if (arg_dry_run)
                 return 0;
 
+        r = sd_bus_message_new_method_call(bus, &m, bus_login_mgr->destination, bus_login_mgr->path,
+                                           bus_login_mgr->interface, actions[a].method_with_flags);
+        if (r < 0)
+                return r;
+
+        SET_FLAG(flags, SD_LOGIND_ROOT_CHECK_INHIBITORS, arg_check_inhibitors);
+
+        r = sd_bus_message_append(m, "t", flags);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_set_allow_interactive_authorization(m, arg_ask_password);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_call(bus, m, 0, &error, NULL);
```

Comment:
Any particular reason for the more verbose set of calls instead of bus_call_method ?

Reply:
Because for WithFlags version, need to set interactive authorization flag to message header. bus_call_method do not support that.

Reply:
this is not necessary. the flag is set globally by default on the whole sd_bus connection object. 

basically: you nowadays don#t have to think about the flag at all anymore.